src/test/kotlin/net/maizegenetics/phgv2/utils/VariantLoadingUtilsTest.kt

```diff
@@ -1,22 +1,129 @@
-        // TBD
+        //Create some simple VariantContext records into a list
+        val refMap = mapOf("1" to NucSeq("AAAAAAAAAAA"), "2" to NucSeq("TTTTTTTTTT"))
+        val variants = listOf(
+            createRefRangeVC(refMap, "Line1", Position("1",1), Position("1",1), Position("1",1), Position("1",1)),
+            createSNPVC("Line1", Position("1",2), Position("1",2), Pair("A","T"), Position("1",2), Position("1",2)),
+            createRefRangeVC(refMap,"Line1", Position("1",3), Position("1",6), Position("1",3), Position("1",6)),
+            createSNPVC("Line1", Position("1",7), Position("1",7), Pair("A","G"), Position("1",2), Position("1",2)),
+            createRefRangeVC(refMap,"Line1", Position("1",8), Position("1",10), Position("1",8), Position("1",10))
+        )
+
+        //Export the variants to a file
+        val testFile = "src/test/kotlin/net/maizegenetics/phgv2/testData/testExportVariantContext.vcf"
+        exportVariantContext("Line1", variants, testFile, refMap, setOf())
+
+        //Load the file back in and check that the variants are as expected
+        val loadedVariants = VCFFileReader(File(testFile), false).iterator().asSequence().toList()
+
+        assertEquals(5, loadedVariants.size, "Unexpected number of variants")
+        for(i in 0 until variants.size) {
+            compareVariants(variants[i], loadedVariants[i])
+        }
+
+    }
+
+    fun compareVariants(expectedVariant: VariantContext, loadedVariant: VariantContext) {
+        //compare the two variant contexts
+        assertEquals(expectedVariant.contig, loadedVariant.contig, "Contig not as expected")
+        assertEquals(expectedVariant.start, loadedVariant.start, "Start position not as expected")
+        assertEquals(expectedVariant.end, loadedVariant.end, "End position not as expected")
+        assertEquals(expectedVariant.genotypes[0].sampleName, loadedVariant.genotypes[0].sampleName, "Sample name not as expected")
+        assertEquals(expectedVariant.alleles[0].baseString, loadedVariant.alleles[0].baseString, "Reference allele not as expected")
+        assertEquals(expectedVariant.alternateAlleles[0].baseString, loadedVariant.alternateAlleles[0].baseString, "Alternate allele not as expected")
+        assertEquals(expectedVariant.getAttributeAsString("ASM_Chr",""), loadedVariant.getAttributeAsString("ASM_Chr",""), "ASM_Chr not as expected")
+        assertEquals(expectedVariant.getAttributeAsInt("ASM_Start",0), loadedVariant.getAttributeAsInt("ASM_Start",0), "ASM_Start not as expected")
+        assertEquals(expectedVariant.getAttributeAsInt("ASM_End",0), loadedVariant.getAttributeAsInt("ASM_End",0), "ASM_End not as expected")
+
     }
 
     @Test
     fun testCreateGenericHeader() {
         // this also tests createGenericHeaderLineSet
+        val taxa = listOf("LineA", "LineB", "LineC")
+        val altHeaders = (1 .. 10).map{VCFAltHeaderLine("<ID=${it}, Description=\"Description for ${it}\">", VCFHeaderVersion.VCF4_2)}.toSet()
```

Comment:
THis will be changed eventually to how the md5 hash as id, correct?  (I guess that is the source code that needs to be whatever we've decided goes into the alt headers)

Reply:
Yes this will be hashes normally.  For the test it just needs to be a String though so it does still work.  Once we have the G/H VCF code up and running that can be used to test storing the hashes here.